lib/awful/rules.lua

```diff
@@ -236,17 +242,67 @@ rules.extra_properties = {}
 --
 -- To add a new properties, just do:
 --
---    function awful.rules.high_priority_properties.my_new_property(c, value)
+--    function awful.rules.high_priority_properties.my_new_property(c, value, props)
 --        -- do something
 --    end
 --
+-- By default, the table has the following functions:
+--
+-- * tag
+--
 -- @tfield table awful.rules.high_priority_properties
+-- @tfield function awful.rules.high_priority_properties.tag FOO
 rules.high_priority_properties = {}
 
 rules.delayed_properties = {
     focus = true,
 }
 
+function rules.high_priority_properties.tag(c, value)
+    if value then
+        if type(value) == "string" then
+            value = atag.find_by_name(nil, value)
+        end
+
+        c:tags{ value }
+    end
+end
+
+function rules.extra_properties.switchtotag(c, value)
+    if not value then return end
+
+    local selected_tags = {}
+
+    for _,v in ipairs(c.screen.selected_tags) do
+        selected_tags[v] = true
+    end
+
+    local tags = c:tags()
+
+    for _, t in ipairs(tags) do
+        t.selected = true
+        selected_tags[t] = nil
+    end
+
+    for t in pairs(selected_tags) do
+        t.selected = false
+    end
+end
+
+function rules.extra_properties.geometry(c, _, props)
+    local cur_geo = c:geometry()
+
+    local new_geo = type(props.geometry) == "function"
+        and props.geometry(c, props) or props.geometry or {}
```

Comment:
Why does this ignore the `value` argument and instead re-computes it? Isn't this line the same as doing `local new_geo = value or {}`?